### ISSUE TITLE
Support for threaded comments

### ISSUE THREAD

**Issue by jenstornell** [Addpixel/KirbyComments#27]
Like nested comments.

**Comment by florianpircher** [comment]
Threaded comments are currently not supported, but once custom fields (see #23) are supported you will be able to implement replies using a custom `reply-to` field which stores the ID of the comment you are replying to. I will add an `nestByField()` method for more convenience in the front-end.

```php
// custom comment rendering code:
function renderComment($comment) {
  echo $comment->message();
  
  if ($comment->hasNestedComments()) {
    foreach ($comment->nestedComments() as $child) {
      renderComment($child);
    }
  }
}

// nest comment by custom field
$nestedComments = $page->comments()->nestByField('reply-to');

foreach ($nestedComments as $comment) {
  renderComment($comment);
}
```

**Comment by florianpircher** [comment]
Comment nesting has been implemented and tested. Shipping soon with v1.5.

<img width="902" alt src="https://user-images.githubusercontent.com/3776198/30277596-9c0680cc-9708-11e7-92f5-295fefd4d98d.png">

**Comment by florianpircher** [comment]
[Version 1.5.0](https://github.com/Addpixel/KirbyComments/releases/tag/1.5.0) is live and on master. You can try out nested comments on the [demo page](https://kirby-comments.addpixel.net/demos/replies).

**Comment by florianpircher** [comment]
For more info have a look at

- the new [`$comments->nestByField()` method](https://github.com/Addpixel/KirbyComments#comments-nestbyfieldreference_field--stringnull-anchor_fieldnull--stringnull-compare_stringvaluetrue--bool--nestedcomment) and
- the new [`NestedComment` class](https://github.com/Addpixel/KirbyComments#nested--nestedcomment-extends-comment).

**Comment by mciszczon** [comment]
Could you please take a look at your demo page for nested comments? It's only showing the Kirby's debug screen (which is not very safe on a public server, by the way).

And while we're at it: documentation on nesting comments is almost non-existent, I can't seem to get it to work. Would it be possible to extend it or at least get a source code of your demo? Then I could figure it out.

Thanks.

**Comment by florianpircher** [comment]
Thank you for reporting the Kirby debug screen! The error has now been fixed (PHP 7.2 comparability issue) and will be included in the next release.

You can download a minimal version of the demo page as an example snippet from the *Replies* demo-page (https://kirby-comments.addpixel.net/demos/replies) and enable it by adding the following code to your config.php:

```php
c::set('comments.custom-fields', array(
	array('name' => 'reply-to')
));
```

Nested comments are implemented using [Custom Fields](https://github.com/Addpixel/KirbyComments#custom-fields) and the [`NestedComment`](https://github.com/Addpixel/KirbyComments#nested--nestedcomment-extends-comment) class. If you have any further questions, feel free to ask them.
